operator/validator/controller.go

```diff
@@ -1050,6 +1041,57 @@ func (c *controller) fetchAndUpdateValidatorsMetadata(logger *zap.Logger, pks []
 	return nil
 }
 
+func (c *controller) ReportValidatorStatuses(ctx context.Context) {
+	ticker := time.NewTicker(time.Second * 30)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			start := time.Now()
+			validatorsPerStatus := make(map[validatorStatus]uint32)
```

Comment:
we should definitely check the resource impact of this loop, we probably can get all these metrics inside the metadata collection loop without running this extra loop.

Reply:
I see what you mean.

The loop in the `UpdateValidatorMetaDataLoop` method iterates over shares via Share Storage, which seems to provide the information we need (_validators with statuses_). This would allow us to avoid iterating over roughly 60k items (validators). Currently, it takes about 7-15 ms, but the concern is not so much the execution time, since it runs in a separate goroutine every 30 seconds, but rather the resource usage.

It seems like a trade-off between reducing resource usage and maintaining cleaner code.

Is it possible to fetch operator's validators and skip iterating over full list of validators? That probably would be the most optimal solution. 

Wdyt @moshe-blox @y0sher ?

Reply:
@oleg-ssvlabs yeah, we made `ValidatorStore` just for that :)

method is called literally OperatorValidators also 🤣 

i think you can already access it from `controller`

Reply:
fixed